baseapp/baseapp.go

```diff
@@ -190,6 +190,9 @@ type BaseApp struct {
 	// by developers.
 	optimisticExec *oe.OptimisticExecution
 
+	// includeNestedMsgsGas holds a set of message types for which gas costs for its nested messages are calculated.
+	includeNestedMsgsGas map[string]struct{}
+
```

Comment:
why don't we calculate nested message cost for all msgs?

Reply:
oh nvm, found answer here: https://github.com/cosmos/cosmos-sdk/pull/20291#discussion_r1598261005

Reply:
this seems confusing. the comment i linked above says we _wouldn't_ want to include the gas for gov, as those messages are queued as more of something like a `system_transaction`, so gas doesn't really matter then. 

so shouldn't this behavior be inverted? i.e. we define a list of messages who's nested messages we _don't_ want to add gas costs for

Reply:
also found this https://github.com/cosmos/cosmos-sdk/pull/20291#discussion_r1677434493.. it seems the rationale was that chain devs may miss this option and not set it for gov.

Reply:
Yeah this seems like a potential footgun either way rn